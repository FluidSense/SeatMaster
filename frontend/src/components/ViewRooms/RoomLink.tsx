import { LenkepanelBase } from 'nav-frontend-lenkepanel';
import { Element } from 'nav-frontend-typografi';
import React from 'react';
import { NavLink } from 'react-router-dom';
import { ROUTE_TO } from './constants';
import { IRoom } from './index';
import { _LINK_NAME_TITLE, _LINK_NOTES_TITLE, _LINK_SEATS_TITLE } from './strings';

interface IProps {
  room: IRoom;
}

const RoomLink: React.FunctionComponent<IProps> = (props) => {
  const { room } = props;
  const seats = room.seats.count !== undefined ? room.seats.count : 0;
  const link = (panelProps: any) => (
    <NavLink
      to={{ pathname: panelProps.href, room: { ...room } }}
      className="lenkepanel lenkepanel--border"
    >
      {panelProps.children}
    </NavLink>);
  return (
    <LenkepanelBase
      key={room.id}
      linkCreator={link}
      href={`${ROUTE_TO}${room.id}`}
    >
      <div className="lenkepanel-rooms">
        <div className="link-name"><Element>{_LINK_NAME_TITLE}</Element>{room.name}</div>
        <div className="link-notes"><Element>{_LINK_NOTES_TITLE}</Element>{room.info}</div>
        <div className="link-seats"><Element>{_LINK_SEATS_TITLE}</Element>{seats}</div>
      </div>
    </LenkepanelBase>
  );
};

export default RoomLink;
