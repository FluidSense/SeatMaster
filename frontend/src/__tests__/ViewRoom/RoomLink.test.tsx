import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import RoomLink from '../../components/ViewRooms/RoomLink';

describe('Renders roomlink correctly', () => {
  const room = { id: 1, name: 'name', info: 'info', seats: { count: 1, seats: [] } };
  it('Renders a room link correctly', () => {
    const wrapper = shallow(<RoomLink room={room} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
