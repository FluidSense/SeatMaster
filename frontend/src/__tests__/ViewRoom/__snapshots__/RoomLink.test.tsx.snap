// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders roomlink correctly Renders a room link correctly 1`] = `
<LenkepanelBase
  border={false}
  href="/admin/rooms/update-room"
  key="1"
  linkCreator={[Function]}
>
  <div
    className="lenkepanel-rooms"
  >
    <div
      className="link-name"
    >
      <Element>
        Name
      </Element>
      name
    </div>
    <div
      className="link-notes"
    >
      <Element>
        Notes
      </Element>
      info
    </div>
    <div
      className="link-seats"
    >
      <Element>
        Number of seats
      </Element>
      1
    </div>
  </div>
</LenkepanelBase>
`;
